syft/frameworks/torch/mpc/przs.py

```diff
@@ -0,0 +1,55 @@
+from collections import defaultdict
+
+import torch
+
+import syft
+from syft.generic.utils import allow_command, remote
+
+RING_SIZE = 2 ** 64
+
+
+def przs_setup(workers):
+    seed_max = 2 ** 32
+    cycle_workers = zip(workers, [*workers[1:], workers[0]])
+
+    workers_ptr = defaultdict(dict)
+
+    for cur_worker, next_worker in cycle_workers:
+        ptr = cur_worker.remote.torch.randint(-seed_max, seed_max - 1, size=(1,))
+        ptr_next = ptr.copy().move(next_worker)
+
+        workers_ptr[cur_worker]["cur_seed"] = ptr
+        workers_ptr[next_worker]["prev_seed"] = ptr_next
+
+    for worker, seeds in workers_ptr.items():
+        cur_seed = seeds["cur_seed"]
+        prev_seed = seeds["prev_seed"]
+        remote(_initialize_generators, location=worker)(cur_seed, prev_seed)
+
+
+@allow_command
+def _initialize_generators(cur_seed, prev_seed):
+    worker = cur_seed.owner
+    cur_generator = torch.Generator()
+    prev_generator = torch.Generator()
+
+    cur_generator.manual_seed(cur_seed.item())
+    prev_generator.manual_seed(prev_seed.item())
+
+    generators = {"przs": {"cur": cur_generator, "prev": prev_generator}}
```

Comment:
does this deserves a class data structure or just a dict?

Reply:
I think we can use a dict for the moment since we only set/get this